contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,74 @@
+# Priority in ResourceQuota
+
+@resouer @bsalamat @davidopp
+
+Aug 2017
+  * [Objective](#objective)
+    * [Non-Goals](#non-goals)
+  * [Background](#background)
+  * [Overview](#overview)
+  * [Detailed Design](#detailed-design)
+    * [Expected behavior of ResourceQuota](#resourcequota)
+    * [Expected behavior of ResourceQuota admission controller](#resourcequota-admission-controller)
+
+
+## Objective
+
+This feature is designed to make ResourceQuota become priority aware, three sub-tasks are included.
+
+* Add a field to ResourceQuota that specifies a priority class name.
+* Incorporate priority in quota checks.
+* ResourceQuota admission controller to check ResourceQuota priority class name and make corresponding decision.
+
+### Non-Goals 
+
+* Add priority in Pod spec (this is implemented in: #45610)
+
+## Background 
+
+Since we already have [priority filed in Pod spec](https://github.com/kubernetes/kubernetes/pull/45610), 
+Pods can now be classified into different priority classes. We would like to be able to create quota for various priority classes in order to manage cluster resources better and limit abuse scenarios. In order to implement this, we need to add a priority class name filed to ResourceQuota definition.
+
+## Overview 
+
+This design doc introduces a new field in ResourceQuota to specify a priority class and explains how quota enforcement logic is changed to apply the quota to pods with the given priority class.
+
+## Detailed Design 
+
+### Expected behavior of ResourceQuota
+
+The design will be like:
+
+```go
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+  // Hard is the set of desired hard limits for each named resource
+  // +optional
+  Hard ResourceList
+  // A collection of filters that must match each object tracked by a quota.
+  // If not specified, the quota matches all objects.
+  // +optional
+  Scopes []ResourceQuotaScope
+  // If specified, indicates this ResourceQuota applies to that priority class
```

Comment:
if specified, the quota's hard limits are restricted to only account resources oriented around a pod (cpu, memory, etc. similar to the behavior with many of our scopes).

i am not sure i agree on the behavior when not specified.  when not specified, the quota should be able account for resources oriented around a pod as well as other resources (services, etc.).  i am concerned that the dominant usage of quota today is that i want to cap the amount of resource this namespace can consume independent of priority.  with this behavior noted here, admins are required to create a quota per priority which will most likely surprise our users during upgrade.